code/game/machinery/doors/airlock.dm

```diff
@@ -500,6 +520,31 @@ About the new airlock wires panel:
 				sleep(6)
 				update_icon(AIRLOCK_CLOSED)
 
+
+/obj/machinery/door/airlock/proc/change_paintjob(obj/item/airlock_painter/W, mob/user)
+	if(!paintable)
+		to_chat(user, "<span class='warning'>You cannot paint this type of airlock.</span>")
+		return FALSE
+
+	if(airlock_material == "glass" && W.paint_setting == "Freezer") // Freezer doesn't have a glass version
+		to_chat(user, "<span class='warning'>The freezer paint job can only be applied to non-glass airlocks.</span>")
+		return FALSE
+
+	if((!in_range(src, user) && loc != user))
+		return FALSE
+
+	if(do_after(user, 20, target = src))
+		W.paint()
+		var/airlock_type = paintable_airlocks["[W.paint_setting]"]
```

Comment:
If you move paintable_airlocks to be paint gun level var, you won't need to worry (as much) about it being desynced from paint gun's own option_list (since you'd be able to construct the latter from the former), and getting a null airlock_type here (which would at best runtime, and at worst make the door invisible).

Oh and (totally a nitpick!) probably a better name would be something like (airlock?)_(available?)_paintjobs, since this, confusingly, has nothing to do with the .paintable airlock var (that one only controls which doors can be changed away *from*, not which the paint gun can *produce*!)

Reply:
Both good points